### DISCUSSION TITLE
Enemy Movement using Object Layer

### DISCUSSION THREAD

**Discussion by GDWR** [jonathan-d-zhang/flawless#35] [Q&A]
In PR #18 a function to load data from an Object Layer was made to allow for enemy spawn and paths to be loaded from map data.

We now need to make the guards use this data to Spawn and Move following the path.

@ForbiddenKJ Please do a small write up on how to create Tiled maps as such to allow for Enemy Spawn and Movement nodes to be read correctly.

**Comment by jbutcher5** [comment]
Yes, i'll get doing that.

**Comment by jbutcher5** [comment]
To create enemy movement using object layers/objects we need to set the way-points. To to this I will be using a program called [Tiled](https://www.mapeditor.org/).

Each enemy's way-points should be on a new layer. This is because it makes it much easier when parsing the xml

**Create a Rectangle**
- In the tool bar hit the "Insert Rectangle" button
![Screenshot from 2021-03-29 21-22-01](https://user-images.githubusercontent.com/36408549/112900709-bb53c900-90db-11eb-9e21-6e53728ac250.png)
- Congratulations! You made a rectangle.

**Creating an Enemy Object Layer**
- Hit the page icon in the bottom left of this image
![Screenshot from 2021-03-29 22-00-32](https://user-images.githubusercontent.com/36408549/112899639-50ee5900-90da-11eb-8343-e9b0b8ead567.png)
- Click object layer
- Name it something appropriate (it doesn't really matter what you choose to name it) in object layer properties
![Screenshot from 2021-03-29 21-22-45](https://user-images.githubusercontent.com/36408549/112899802-85faab80-90da-11eb-9b32-8e8658b2ac6d.png)
- Get adding a spawn and points.

**Creating a Spawn**

- Create a rectangle
- Change the rectangles type to spawn and name it what ever you want, something along the lines of "guard1_spawn"
- Change the width and height to match tile map (8x8)
- **Ctrl + Drag, (THIS SNAPS IT TO GRID AND MOVES IT, THIS MUST BE USED)**  to set the position
![Screenshot from 2021-03-29 21-23-07](https://user-images.githubusercontent.com/36408549/112897365-7037b700-90d7-11eb-86ff-17a0b40738c4.png)

**Creating a Point**

- Create a rectangle
- Change the rectangles type to point
- Change the rectangles name to order of which you would like the enemy to visit it, for the first tile I would like it to walk to I would name it "0" and for the second "1" and so on, **IT MUST START AT 0**.
- Change the width and height to match tile map (8x8)
- **Ctrl + Drag, (THIS SNAPS IT TO GRID AND MOVES IT, THIS MUST BE USED)**  to set the position
![Screenshot from 2021-03-29 21-23-22](https://user-images.githubusercontent.com/36408549/112898113-59459480-90d8-11eb-8fad-9d1f8f0d935e.png)

**Product**
Your final product must look something like this.
![Screenshot from 2021-03-29 22-24-14](https://user-images.githubusercontent.com/36408549/112902110-94969200-90dd-11eb-9de4-1a7cfdcb0972.png)
